### DISCUSSION TITLE
Where are the prompt generation models saved to?

### DISCUSSION THREAD

**Discussion by illtellyoulater** [adieyal/sd-dynamic-prompts#206] [Q&A] [answered]
Does anybody know where the models are saved?
I thought they were in a subdirectory of the extension, but there are no models there...?

**Comment by adieyal** [comment]
I don't understand what you mean by prompt generation models, could you clarify?

**Comment by adieyal** [comment, marked as answer]
Do you mean for Magic Prompt? If so, on my machine they are saved in `$HOME/.cache/huggingface/hub` - the transformers library automatically downloads them

**Reply by illtellyoulater** [reply]
ah ok, found them! thanks!

**Comment by Woisek** [comment]
What path would this be on a Windows machine ... ?

**Reply by illtellyoulater** [reply]
c:\users\username\\.cache\huggingface\hub\

**Comment by Woisek** [comment]
First ... thanks, I found it.
Second ... wowsers!! Why is this on the C drive? What program has it set to there? No wonder my system backups have grown so big unnatural lately.
Thirdly: Can this path be changed via configuration? This cache folder is 20GB in size and has to be moved away from C.

**Comment by adieyal** [comment]
The models are downloaded by Hugging Face's transformers library. You can change the directory by setting the [HF_DATASETS_CACHE](https://huggingface.co/docs/datasets/cache#cache-directory) environment variable. I'm not familiar with windows but some quick Googling will help you figure out where to set it. 

Please report back when you managed to change it for the benefit of other users.

**Reply by Woisek** [reply]
Thanks for the  hint. As far as I can see, this only applies to the huggingface part. What about the clip and the torch folders in that same .cache folder? They are 1.52 and 1.86GB. They have to move, also.
Any hints on that, too?
Thanks for your help.

**Comment by adieyal** [comment]
Not sure - I don't think those were installed by Dynamic Prompts.

**Reply by Woisek** [reply]
Hmm ... OK, have to look further. But I guess it will likely to be, that I "junction out" the folder ...
